src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -111,9 +112,23 @@ export class ExtensionTipsService implements IExtensionTipsService {
 		this.importantRecommendationsIgnoreList = <string[]>JSON.parse(this.storageService.get('extensionsAssistant/importantRecommendationsIgnore', StorageScope.GLOBAL, '[]'));
 
 		// retrieve ids of previous recommendations
-		const storedRecommendations = <string[]>JSON.parse(this.storageService.get('extensionsAssistant/recommendations', StorageScope.GLOBAL, '[]'));
-		for (let id of storedRecommendations) {
-			this._recommendations[id] = true;
+		const storedRecommendationsJson = JSON.parse(this.storageService.get('extensionsAssistant/recommendations', StorageScope.GLOBAL, '[]'));
+		if (Array.isArray<string>(storedRecommendationsJson)) {
+			for (let id of <string[]>storedRecommendationsJson) {
+				this._recommendations[id] = Date.now();
+			}
+		} else {
+			const now = Date.now();
+			forEach(storedRecommendationsJson, entry => {
+				if (typeof entry.value === 'number') {
+					const diff = (now - entry.value) / milliSecondsInADay;
+					if (diff > 7) {
+						delete this._recommendations[entry.value];
```

Comment:
Does this should be 

` delete this._recommendations[entry.key]` 

May be its the time to change the type of `this._recommendations` to `Map`?

Reply:
Generally I'd say but if this comes from and goes to JSON it might be easier to stick to an old object?